keel/api/v1/cases/views.py

```diff
@@ -81,3 +81,42 @@ def get_case(self, request, **kwargs):
         
         response["message"] = data
         return Response(response)
+
+class UpdateCaseProgramView(GenericViewSet):
+    serializer_class = CaseProgramSerializer
+    permission_classes = (permissions.IsAuthenticated, )
+    authentication_classes = (JWTAuthentication, )
+
+    def get_all_programs(self, request):
+        response = {'status':1, 'message':'All programs retreived', 'data':''}
+        queryset = Program.objects.all()
+        serializer = BaseCaseProgramSerializer(queryset, many=True)
+        response['data'] = serializer.data
+        return Response(response)
+
+    def update_program(self, request, **kwargs):
+        response = {'status':1, 'message':'Program Updated successfully', 'data':''}
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        program = serializer.validated_data
+        case_id = kwargs.get('case_id')
+
+        try:
+            case = Case.objects.get(case_id=case_id)
+            case.program = program
+            case.save()
+        except Case.DoesNotExist as e:
+            logger.error('ERROR: CASE:UpdateCaseProgramView ' + str(e))
+            response['message'] = str(e)
+            response['status'] = 0
+            return Response(response, status=status.HTTP_404_NOT_FOUND)
+        except Exception as e:
+            logger.error('ERROR: CASE:UpdateCaseProgramView ' + str(e))
+            response['message'] = str(e)
+            response['status'] = 0
+            return Response(response, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
this shouldnt be a 400...but 500 instead...

Reply:
fixed